Extensions/titles.js

```diff
@@ -0,0 +1,104 @@
+//* TITLE Tab titles **//
+//* VERSION 1.0.0 **//
+//* DESCRIPTION Descriptive tab titles, rather than just “Tumblr” **//
+//* DEVELOPER Rebecca Turner **//
+//* FRAME false **//
+//* BETA false **//
+
+XKit.extensions.titles = new Object({
+	running: false,
+
+	/* object mapping the first url folder component to a function which takes
+	 * an array of url components (like ['blog', 'xkit'] for '/blog/xkit') and
+	 * returns the new page title
+	 */
+	titleFuncs: {
+		blog: url => {
+			if (url.length > 2) {
+				if (url[2] == "new") {
+					// new post
+					return "Post to " + url[1];
+				} else if (url[2] == "review") {
+					return url[1] + "’s flagged posts";
+				} else if (url[2] != "delete") {
```

Comment:
this logic is backwards, and it makes it harder to read. it should be:

```js
} else if (url[2] == "delete") {
  return "Delete " + url[1];
} else {
  return url[1] + "’s " + url[2];
}
```

but I would just as soon turn this into an object like below:

```js
const [blogName, route] = url;
return {
  "new": `Post to ${blogName}`,
  review: `${blogName}'s flagged posts`,
  delete: `Delete ${blogName}`
}[route] || blogName
```

Reply:
Turned into an object.